spaceship/lib/spaceship/client.rb

```diff
@@ -780,7 +782,9 @@ def log_request(method, url, params, headers = nil, &block)
       params_to_log = params_to_log.collect do |key, value|
         "{#{key}: #{value}}"
       end
-      logger.info(">> #{method.upcase} #{url}: #{body.to_json} #{params_to_log.join(', ')}")
+      if log_body
+        logger.info(">> #{method.upcase} #{url}: #{body.to_json} #{params_to_log.join(', ')}")
```

Comment:
We definitely need an `else` here that still logs the request, but without the body.

Alternatively we could rewrite this so a new `body_to_log` variable is introduced with value `""` before the `if body`, then set to `body.to_json` if we could parse it to JSON, and to `"[body could not be encoded for logging]"` in the `rescue JSON::ParserError`... then the actual `logger.info` stays simple.

Reply:
I created my suggestion as an individual PR at https://github.com/fastlane/fastlane/pull/13825